From 0a617e9e9902e5986c8835e2e466e581d853b4f5 Mon Sep 17 00:00:00 2001
From: Juha Kallioinen <juha.kallioinen@jollamobile.com>
Date: Fri, 13 Sep 2013 19:47:31 +0000
Subject: [PATCH] [sb2] Delay creation of run symlink creation for /var/run

A link to target_root/run is needed from /tmp/sbox_session_dir/run,
which will be where /var/run symlink points. The symlink was created
too early and ended up pointing to the host's /run directory. This
ended up causing problems in sb2 path resolution.

Signed-off-by: Juha Kallioinen <juha.kallioinen@jollamobile.com>
---
 utils/sb2 | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/utils/sb2 b/utils/sb2
index 2002739..f1a68c7 100755
--- a/utils/sb2
+++ b/utils/sb2
@@ -241,6 +241,13 @@ function load_configuration()
 		SBOX_TOOLS_ROOT=`readlink -f $SBOX_TOOLS_ROOT`
 	fi
 
+	# now that we know SBOX_TOOLS_ROOT, we can create the required
+	# link for the /var/run symlink. it might already be there if
+	# this is a restored session
+	if [ ! -L "$SBOX_SESSION_DIR/run" ]; then
+	    ln -s $SBOX_TARGET_ROOT/run $SBOX_SESSION_DIR/run
+	fi
+
 	# older sb2.config files had values for the Debian
 	# variables; unset them (this is necessary only if
 	# the target still has an old-format sb2.config)
@@ -1405,8 +1412,12 @@ function initialize_new_sb2_session()
 	# conflicts with SB2's symlink resolution algorithm,
 	# because e.g. "emulate" mode has a separate rule for
 	# /var/run. But yet another symlink in the session
-	# directory helps:
-	ln -s $SBOX_TARGET_ROOT/run $SBOX_SESSION_DIR/run
+	# directory helps...
+	#
+	# Alas! at this time we don't yet know SBOX_TARGET_ROOT, so
+	# this symlink can earliest be created at load_configuration()
+	#
+	# ln -s $SBOX_TARGET_ROOT/run $SBOX_SESSION_DIR/run
 
 	# ruletree ipc client sockets.
 	mkdir $SBOX_SESSION_DIR/sock
-- 
1.8.3-rc3

